lib/ansible/modules/cloud/pubnub/pubnub_blocks.py

```diff
@@ -0,0 +1,2373 @@
+  pubnub_blocks:
+    email: '{{ email }}'
+    password: '{{ password }}'
+    application: '{{ app_name }}'
+    keyset: '{{ keyset_name }}'
+    name: '{{ block_name }}'
+    event_handlers:
+      -
+        name: '{{ handler_name }}'
+        event: 'js-after-publish'
+
+# Stop block and event handlers.
+- name: Stopping block
+  pubnub_blocks:
+    email: '{{ email }}'
+    password: '{{ password }}'
+    application: '{{ app_name }}'
+    keyset: '{{ keyset_name }}'
+    name: '{{ block_name }}'
+    state: stop
+
+# Multiple module calls with cached result passing
+- name: Create '{{ block_name }}' block
+      register: module_cache
+      pubnub_blocks:
+        email: '{{ email }}'
+        password: '{{ password }}'
+        application: '{{ app_name }}'
+        keyset: '{{ keyset_name }}'
+        name: '{{ block_name }}'
+        state: present
+    - name: Add '{{ event_handler_1_name }}' event handler to '{{ block_name }}'
+      register: module_cache
+      pubnub_blocks:
+        cache: '{{ module_cache }}'
+        application: '{{ app_name }}'
+        keyset: '{{ keyset_name }}'
+        name: '{{ block_name }}'
+        state: present
+        event_handlers:
+          -
+            src: '{{ path_to_handler_1_source }}'
+            name: '{{ event_handler_1_name }}'
+            channels: '{{ event_handler_1_channel }}'
+            event: 'js-before-publish'
+    - name: Add '{{ event_handler_2_name }}' event handler to '{{ block_name }}'
+      register: module_cache
+      pubnub_blocks:
+        cache: '{{ module_cache }}'
+        application: '{{ app_name }}'
+        keyset: '{{ keyset_name }}'
+        name: '{{ block_name }}'
+        state: present
+        event_handlers:
+          -
+            src: '{{ path_to_handler_2_source }}'
+            name: '{{ event_handler_2_name }}'
+            channels: '{{ event_handler_2_channel }}'
+            event: 'js-before-publish'
+    - name: Start '{{ block_name }}' block
+      register: module_cache
+      pubnub_blocks:
+        cache: '{{ module_cache }}'
+        application: '{{ app_name }}'
+        keyset: '{{ keyset_name }}'
+        name: '{{ block_name }}'
+        state: start
+'''
+
+RETURN = '''
+module_cache:
+  description: "Cached account information. In case if with single play module used few times it is better to pass cached data to next module calls to speed up
+  process."
+  type: dict
+'''
+
+PUBNUB_API_URL = "https://admin.pubnub.com/api"
+PN_BLOCK_STATE_CHECK_INTERVAL = 1
+PN_BLOCK_STATE_CHECK_MAX_COUNT = 30
+# Import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import *
+from ansible.module_utils.urls import *
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+try:
+    import zlib
+    ZLIB_AVAILABLE = True
+except ImportError:
+    ZLIB_AVAILABLE = False
+import random
+import copy
+import time
+
+
+class PubNubAccount(object):
```

Comment:
Is there no existing Python library to manage BLOCKS API logins & resources? Ansible is ok with modules that require an external client (such as the `boto3` client for AWS APIs, or `shade` for OpenStack). If there's an equivalent for PubNub it would be great not to duplicate code here.